src/xc_integrator/local_work_driver/device/local_device_work_driver.hpp

```diff
@@ -61,35 +61,26 @@ class LocalDeviceWorkDriver : public LocalWorkDriver {
   void eval_collocation_gradient( XCDeviceData* );
   void eval_collocation_hessian( XCDeviceData* );
 
-  void eval_xmat( double fac, XCDeviceData*, bool do_grad = false );
-
-  void eval_uvvar_lda_rks( XCDeviceData* );
-  void eval_uvvar_gga_rks( XCDeviceData* );
-
-  void eval_uvvar_lda_uks( XCDeviceData* );
-  void eval_uvvar_gga_uks( XCDeviceData* );
-
-  void eval_uvvar_lda_gks( XCDeviceData* );
-  void eval_uvvar_gga_gks( XCDeviceData* );
+  void eval_xmat( double fac, XCDeviceData*, bool do_grad = false , density_id den = DEN_S );
```

Comment:
since this function has become non-trivial, best to remove the defaults here and be explicit wherever its used